docs/src/modules/spring/pages/value-entity.adoc

```diff
@@ -1 +1,112 @@
+
+Having a good idea of what we want to build and how we will interact with it, we will need to:
+
+- declare your entity and pick an entity key (it needs to be unique as it will be used for sharding purposes);
+- define an access point (i.e. a route path) to your entity;
+- implement how each command is handled.
+
+Let us start by showing how to create the Value Entity:
+
+.src/main/java/com/example/CounterEntity.java
+[source,java]
+----
+include::example$spring-valueentity-counter/src/main/java/com/example/CounterEntity.java[tags=declarations]
+----
+<1> Every Entity must be annotated with `@EntityType` with a stable name. This name should be unique among the
+different existing entities within a Kalix application.
+<2> The `@EntityKey` value should be unique per entity and map to some field being received on the route path, in this example it's the `counter_id`.
+<3> The `CounterEntity` class should extend `kalix.javasdk.valueentity.ValueEntity`.
+<4> The initial state of each counter is defined as 0.
+
+=== Updating state
+
+We will now show how to add the command handlers for supporting the two desired operations (`set` and `plusone`). Command handlers are implemented as methods on the entity class but are also exposed for external interactions and always return an `Effect` of some type.
+
+.src/main/java/com/example/CounterEntity.java
+[source,java,indent=0]
+----
+include::example$spring-valueentity-counter/src/main/java/com/example/CounterEntity.java[tags=behaviour]
+----
+<1> Expose the `set` command handler on path `/counter/\{id\}/set` as a PUT endpoint where `id` will be its unique identifier.
+<2> Set the new counter value to the value received as body of the command request.
+<3> Reply with the new counter value wrapped within a `Number` object.
+<4> The method is accessible as a POST endpoint on `/counter/\{id\}/plusone`, where `id` will be its unique identifier.
+<5> Its `plusone` increases the counter by adding 1 to the current state.
+<6> Finally, using the Effect API, we instruct Kalix to persist the new state, and we build a reply with the wrapper object.
+
+NOTE: The `id` parameter matches the `entityKey` value. Also, for this example, we have opted to always repeat the common route `/counter/\{id\}` for each command but a simpler option could be to use a `@RequestMethod("/counter/\{id\}")` at class level.
+
+=== Retrieving state
+
+The following example shows how to implement a simple endpoint to retrieve the current state of the entity, in this case the value for a specific counter.
+
+.src/main/java/com/example/CounterEntity.java
+[source,java,indent=0]
+----
+include::example$spring-valueentity-counter/src/main/java/com/example/CounterEntity.java[tags=query]
+----
+<1> Expose the `get` query handler on path `/counter/\{id\}` as a GET endpoint where `id` will be its unique identifier.
+<2> Reply with the current state wrapped within a `Number`.
+
+== Testing the Entity
+
+include::partial$testing-entity.adoc[]
```

Comment:
This partial has the text in the bullets starting in lowercase

Reply:
fixed now